### PR TITLE
BP-933 No second chance email sended on pending order

### PR DESCRIPTION
**Pull request by saranchuk-hys**

### CODE REVIEW

File: Controller/Payconiq/Process.php [hunk 1]
```diff
@@ -164,4 +176,15 @@ protected function getList()
 
         return $list;
     }
+
+    public function secondChance($order){
```
Comment (florinel-chis) [inline Controller/Payconiq/Process.php:180]: this should be part of a second chance service class. Do not use/add additional methods, classes and logic in every place a second chance record needs to be created.

Comment (florinel-chis) [reply Controller/Payconiq/Process.php:180]: and the same service class should be used here - https://github.com/buckaroo-it/Magento2/blob/760ded2df4800cb473c4e531b361edc70be618f5/Controller/Redirect/Process.php#L717 (which is almost identical to this one, except in there order is retrieved from the class, here it's passed as argument)

Comment (saranchuk-hys) [reply Controller/Payconiq/Process.php]: fixed done

File: Controller/Payconiq/Process.php [hunk 2]
```diff
@@ -67,7 +70,10 @@ public function __construct(
         TransactionRepositoryInterface $transactionRepository,
         Account $account,
         TransactionCancel $transactionCancel,
-        QuoteRecreate $quoteRecreate
+        QuoteRecreate $quoteRecreate,
+        \Buckaroo\Magento2\Model\SecondChanceRepository $secondChanceRepository,
+        \Magento\Framework\Math\Random $mathRandom,
```
Comment (florinel-chis) [inline Controller/Payconiq/Process.php:75]: not used I assume, can be removed

File: Controller/Payconiq/Process.php [hunk 3]
```diff
@@ -67,7 +70,10 @@ public function __construct(
...
+        \Magento\Framework\Stdlib\DateTime\DateTime $dateTime
```
Comment (florinel-chis) [inline Controller/Payconiq/Process.php:76]: not used I assume, can be removed

File: Model/SecondChanceRepository.php [hunk 4]
```diff
@@ -187,10 +272,249 @@ public function delete(SecondChanceInterface $SecondChance)
...
+            'billing'                  => $order->getBillingAddress(),
+            'payment_html'             => $this->getPaymentHtml($order),
+            'store'                    => $store,
+            'formattedShippingAddress' => $this->getFormattedShippingAddress($order),
+            'formattedBillingAddress'  => $this->getFormattedBillingAddress($order),
+            'secondChanceToken'        => $secondChance->getToken(),
+        ];
+
+        if ($step == 1) {
+            $templateId = $this->scopeConfig->getValue($config['template'], ScopeInterface::SCOPE_STORE, $store) ??
```
Comment (florinel-chis) [inline Model/SecondChanceRepository.php:418]: I've see a config provider added, that should be used here too. and scopeconfig could be removed in this class.

File: Controller/Payconiq/Process.php (hunk 2 above)
Comment (saranchuk-hys) [reply Controller/Payconiq/Process.php]: Sorry, hover SecondChanceRepository and QuoteRecreate used, mathRandom and DateTime - removed

File: Controller/Payconiq/Process.php (hunk 3 above)
Comment (saranchuk-hys) [reply Controller/Payconiq/Process.php]: Sorry, hover SecondChanceRepository and QuoteRecreate used, mathRandom and DateTime - removed

**Review by florinel-chis — APPROVED** [review]

File: Model/SecondChanceRepository.php (hunk 4 above)
Comment (saranchuk-hys) [reply Model/SecondChanceRepository.php]: fixed

**Review by florinel-chis — APPROVED** [review]

**Pull request merged**